common/src/main/java/com/wynntils/wc/model/Character.java

```diff
@@ -90,17 +104,32 @@ public int getLevel() {
             return level;
         }
 
-        public UUID getId() {
+        public int getId() {
             return id;
         }
 
-        public static CharacterInfo parseCharacter(ItemStack itemStack) {
+        public static CharacterInfo parseCharacter(ItemStack itemStack, int slotNum) {
             List<String> lore = ItemUtils.getLore(itemStack);
-            for (String s : lore) {
-                // Reference.LOGGER.info("Lore: " + s);
+
+            int level = 0;
+            ClassType classType = null;
+
+            for (String line : lore) {
+                Matcher matcher = LEVEL_PATTERN.matcher(line);
+                if (matcher.matches()) {
+                    level = Integer.parseInt(matcher.group(1));
+                    continue;
+                }
+
+                matcher = CLASS_PATTERN.matcher(line);
+
+                if (matcher.matches()) {
+                    classType = ClassType.fromName(matcher.group(1));
+                }
             }
 
-            return new CharacterInfo(null, false, 0, UUID.randomUUID());
```

Comment:
Oh, I never finished this! 😊 

I could have sworn I did a try-to-find-best-matching character, and if none were found, return its UUID. But apparently I only had the design in my head but no implementation. :(

I should get back to this and try to get a implementation that is as robust as we can get, and which supports autojoin etc. I believe we can get it pretty reliable.